pkg/proxy/iptables/proxier_test.go

```diff
@@ -1345,6 +1347,423 @@ func assertIPTablesRulesNotEqual(t *testing.T, expected, result string) {
 	}
 }
 
+// ruleMatchesIP helps test whether an iptables rule such as "! -s 192.168.0.0/16" matches
+// ipStr. ruleAddress is either an IP address ("1.2.3.4") or a CIDR string
+// ("1.2.3.0/24"). negated is whether the iptables rule negates the match.
+func ruleMatchesIP(negated bool, ruleAddress, ipStr string) bool {
+	ip := netutils.ParseIPSloppy(ipStr)
+	if ip == nil {
+		panic("bad IP in test case: " + ipStr)
+	}
+
+	var matches bool
+	if strings.Contains(ruleAddress, "/") {
+		_, cidr, err := netutils.ParseCIDRSloppy(ruleAddress)
+		if err != nil {
+			panic("bad CIDR in kube-proxy output: " + err.Error())
+		}
+		matches = cidr.Contains(ip)
+	} else {
+		ip2 := netutils.ParseIPSloppy(ruleAddress)
+		if ip2 == nil {
+			panic("bad IP/CIDR in kube-proxy output: " + ruleAddress)
+		}
+		matches = ip.Equal(ip2)
+	}
+	return (!negated && matches) || (negated && !matches)
+}
+
+var addChainRegex = regexp.MustCompile(`-A ([^ ]*) `)
+var commentRegex = regexp.MustCompile(`-m comment --comment ("[^"]*"|[^" ]*) `)
+var moduleRegex = regexp.MustCompile("-m ([^ ]*)")
+var localRegex = regexp.MustCompile("(!)? --(src|dst)-type LOCAL")
+var destIPRegex = regexp.MustCompile("(!)? -d ([^ ]*) ")
+var destPortRegex = regexp.MustCompile(" --dport ([^ ]*) ")
+var sourceIPRegex = regexp.MustCompile("(!)? -s ([^ ]*) ")
+var affinityRegex = regexp.MustCompile(" --rcheck ")
+var jumpRegex = regexp.MustCompile("(--probability.*)? -j ([^ ]*)( --to-destination (.*))?$")
```

Comment:
In this case it couldn't be reordered. (The overall syntax is `[match clauses...] [-j target [target options...]]`.)

But if you could reorder the clauses, and someone did, then the patterns wouldn't match any more and a bunch of tests would fail...

Reply:
ehhhhh ok.  I'm a big fan of comments to leave these sorts of clues for future-me :)